Simplify schema

roberthorn opened the pull request.

pkmitre committed:
Remove unnecessary fields

roberthorn committed:
added NPPES data importer

- added rake tasks to execute importer for full and update files
- added changes to importer to match schema changes

pkmitre commented:
I've tested the importer, and there's a speed issue. Importing just 50K entries, the re-written importer takes almost 11 minutes, compared to 15 seconds for the old importer (over 40x slower). It won't be feasible to use this on the full file, with ~4.5 million entries.

roberthorn committed:
fixed performance issue with importer

roberthorn committed:
removed debugging statements

roberthorn commented:
I've fixed the performance issue with the importer. The problem was with a late change I made to how the entity ids were prefetched, I thought it wouldn't have such a negative impact, bad assumption on my part. The base import is back to normal speeds, I imported 100k records under a minute. The update function still takes a bit longer since it's having to check for existing records and a few more operations and ran for ~15 minutes with ~16k records.

roberthorn committed:
changed id prefetching

now we fetch the next value for the sequence ids at the
beginning and increment them wihtin the code instead of
doing a DB query each time. this is done both for full imports
and updates.
